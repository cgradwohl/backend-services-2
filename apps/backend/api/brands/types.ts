import { ApiRequestContext } from "~/lib/lambda-response";
import {
  IApiBrandsGetResponse,
  IApiBrandsListResponse,
  IApiBrandsPostRequest,
  IApiBrandsPostResponse,
  IApiBrandsPutRequest,
  IApiBrandsPutResponse,
} from "~/types.public";

export type ApiBrandsResponse =
  | IApiBrandsGetResponse
  | IApiBrandsListResponse
  | IApiBrandsPostResponse
  | IApiBrandsPutResponse;

export type GetFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiBrandsGetResponse;
}>;

export type ListFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiBrandsListResponse;
}>;

export type PostRequestBody = IApiBrandsPostRequest;

export type PostFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiBrandsPostResponse;
}>;

export type PutRequestBody = IApiBrandsPutRequest;

export type PutFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiBrandsPutResponse;
}>;

export type RemoveFn = (
  context: ApiRequestContext
) => Promise<{
  status: 204;
}>;
