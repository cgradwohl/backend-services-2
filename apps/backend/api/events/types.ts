import { ApiRequestContext } from "~/lib/lambda-response";
import {
  IApiEventsGetResponse,
  IApiEventsListResponse,
  IApiEventsPutRequest,
  IApiEventsPutResponse,
} from "~/types.public";

export type ApiEventsResponse = IApiEventsGetResponse | IApiEventsListResponse;

export type GetFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiEventsGetResponse;
}>;

export type ListFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiEventsListResponse;
}>;

export type PutRequestBody = IApiEventsPutRequest;

export type PutFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiEventsPutResponse;
}>;

export type RemoveFn = (
  context: ApiRequestContext
) => Promise<{
  status: 204;
}>;
