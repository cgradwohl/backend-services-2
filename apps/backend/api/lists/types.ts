import { ApiRequestContext, Response } from "~/lib/lambda-response";

import {
  IApiDeleteListItemResponse,
  IApiGetListItemResponse,
  IApiGetListItemsResponse,
  IApiGetListItemSubscriptionsResponse,
  IApiListItemSubscribeResponse,
  IApiListItemUnsubscribeResponse,
  IApiPutListItemResponse,
  IApiRestoreListItemResponse,
} from "~/types.public";

export type HttpEventHandler<T extends Response<any>> = (
  context: ApiRequestContext
) => Promise<T>;

export type ListApiResponse =
  | IApiDeleteListItemResponse
  | IApiGetListItemResponse
  | IApiGetListItemsResponse
  | IApiGetListItemSubscriptionsResponse
  | IApiListItemSubscribeResponse
  | IApiListItemUnsubscribeResponse
  | IApiPutListItemResponse
  | IApiRestoreListItemResponse;
