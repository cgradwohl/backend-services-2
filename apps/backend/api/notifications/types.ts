import { ApiRequestContext } from "~/lib/lambda-response";
import {
  IApiNotificationGetResponse,
  IApiNotificationGetSubmissionChecksResponse,
  IApiNotificationListResponse,
  IApiNotificationPutSubmissionChecksResponse,
} from "~/types.public";

export type ApiNotificationResponse =
  | IApiNotificationGetResponse
  | IApiNotificationListResponse
  | IApiNotificationGetSubmissionChecksResponse
  | IApiNotificationPutSubmissionChecksResponse;

export type GetFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiNotificationGetResponse;
}>;

export type PostFn = (
  context: ApiRequestContext
) => Promise<{
  status: number;
}>;

export type PutFn = (
  context: ApiRequestContext
) => Promise<{
  status: number;
}>;

export type ListFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiNotificationListResponse;
}>;

export type GetSubmissionChecksFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiNotificationGetSubmissionChecksResponse;
}>;

export type PutSubmissionChecksFn = (
  context: ApiRequestContext
) => Promise<{
  body: IApiNotificationPutSubmissionChecksResponse;
}>;

export type DeleteSubmissionChecksFn = (
  context: ApiRequestContext
) => Promise<{
  status: 204 | 409;
}>;
