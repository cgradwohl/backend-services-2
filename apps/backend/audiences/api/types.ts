import { ApiRequestContext } from "~/lib/lambda-response";
import { FilterConfig } from "~/audiences/stores/dynamo/types";

export type Resource =
  | "/audiences"
  | "/audiences/{audience_id}"
  | "/audiences/{audience_id}/members"
  | "/audiences/{audience_id}/members/{member_id}";

export type MemberResource = "/members/{member_id}/audiences";

export type BaseAudienceResponse = {};

export interface AudienceHandler<T extends BaseAudienceResponse> {
  (context: ApiRequestContext): Promise<T>;
}

export type Audience = {
  created_at: string;
  description: string;
  id: string;
  name: string;
  // TODO: https://linear.app/trycourier/issue/C-5533/rule-engine-ux-for-allowing-people-to-create-audiences-rule
  // rules will never be null, but until we have a UI for creating audiences, we need to allow it to be null
  filter: FilterConfig | null;
  updated_at: string;
};

export type AudienceMember = {
  added_at: string;
  audience_id: string;
  audience_version: number;
  member_id: string;
  reason: string;
};

export interface AudienceListResponse extends BaseAudienceResponse {
  body: {
    paging: {
      cursor: string;
      more: boolean;
    };
    items: Audience[];
  };
}

export interface AudienceDeleteResponse extends BaseAudienceResponse {
  status: 204;
}

export interface AudienceGetResponse extends BaseAudienceResponse {
  status: 200;
  body: Audience;
}

export interface AudienceMemberGetResponse extends BaseAudienceResponse {
  status: 200;
  body: AudienceMember;
}

export interface AudienceMemberListResponse extends BaseAudienceResponse {
  status: 200;
  body: {
    items: AudienceMember[];
    paging: {
      cursor: string;
      more: boolean;
    };
  };
}

export interface AudiencePutResponse extends BaseAudienceResponse {
  status: 200;
  body: {
    audience: Audience;
  };
}

export interface AudienceMemberPutResponse extends BaseAudienceResponse {
  status: 200;
  body: {
    audience_member: AudienceMember;
  };
}

export type Method = "get" | "put" | "delete";

export type UserMethod = "get";
