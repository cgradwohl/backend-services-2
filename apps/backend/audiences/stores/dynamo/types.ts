import { Operator } from "~/audiences/lib/audience-rule-engine";
export interface IDDBAudienceKeys {
  gsi1pk: string;
  pk: string;
  shardId?: number;
  gsi2pk?: string;
}

type Without<T, U> = { [P in Exclude<keyof T, keyof U>]?: never };
type XOR<T, U> = T | U extends object
  ? (Without<T, U> & U) | (Without<U, T> & T)
  : T | U;

interface BaseFilterConfig {
  operator: Operator;
}

export interface SingleFilterConfig extends BaseFilterConfig {
  path: string;
  value: string | boolean;
}

interface NestedFilterConfig extends BaseFilterConfig {
  filters: FilterConfig[];
}

export type FilterConfig = XOR<SingleFilterConfig, NestedFilterConfig>;

export interface IDDBAudience extends IDDBAudienceKeys {
  audienceId: string;
  createdAt: string;
  description?: string;
  lastSendAt?: string;
  memberCount?: number;
  name?: string;
  updatedAt: string;
  version: number;
  filter: FilterConfig;
  workspaceId: string;
}

export interface IDDBAudienceMember extends IDDBAudienceKeys {
  addedAt: string;
  audienceId: string;
  audienceVersion: number;
  userId: string;
  reason: string;
  workspaceId: string;
}

export interface IDDBAudienceCalculation extends IDDBAudienceKeys {
  audienceId: string;
  lastUpdatedAt: string;
  result: "calculating" | "calculated" | "error";
  totalUsers: number;
  totalUsersFiltered: number;
  userCount: number;
  workspaceId: string;
}
