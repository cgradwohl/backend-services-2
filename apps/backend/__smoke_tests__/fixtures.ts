const _emailEvents = (() => {
  if (process.env.SMOKE_TEST_EVENT) {
    return [process.env.SMOKE_TEST_EVENT];
  }

  if (process.env.STAGE === "staging") {
    return ["SAPBJ1JWR44V2MQ3NBRFRVAC594W"];
  }
  if (process.env.STAGE === "production") {
    return ["14171CNHXG4JHXNH2NRMBYJ0Y20P"];
  }

  return [];
})();

export const emailEvents = _emailEvents;
