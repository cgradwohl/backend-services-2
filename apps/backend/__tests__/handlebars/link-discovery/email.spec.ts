import getProviderTemplate from "~/handlebars/template";
import { TemplateConfig } from "~/handlebars/template/types";
import hydrateBlock from "~/lib/blocks/hydrate-slate-block";
import createLinkHandler, { ILinkData } from "~/lib/link-handler";
import createVariableHandler from "~/lib/variable-handler";

import fixtures from "../__fixtures__";
import mockData from "../template/util/mock-data";
import mockProfile from "../template/util/mock-profile";
import getTrackingHandler from "./util/get-tracking-handler";

let usingTemplate = false;
jest.mock("~/handlebars/partials", () => {
  const partials = jest.requireActual("~/handlebars/partials");
  const courierTemplate = partials.default.email["courier-template"];

  // going to dynamically render the template based on usingTemplate
  Object.defineProperty(partials.default.email, "courier-template", {
    get: () => (usingTemplate ? courierTemplate : undefined),
  });

  return partials;
});

const { blocks: blockFixtures, templates: templateFixtures } = fixtures;

const hydrateBlocks = (blocks: any[]) => blocks.map(hydrateBlock);

describe("email blocks", () => {
  const testBlocks = [
    "actionBlock",
    "actionBlockLink",
    "actionBlockWebhook",
    "imageBlock",
    "imageBlockWithVars",
    "listBlockImagesWithLinks",
    "quoteBlockWithLink",
    "templateBlockWithLinks",
    "textBlockWithLink",
    "textBlockWithLinks",
  ];

  const only = false; // "actionBlock";

  beforeEach(() => (usingTemplate = false));

  testBlocks.forEach((block) => {
    const fn = only && only === block ? it.only : it;

    fn(`should render ${block}`, () => {
      const allBlocks = hydrateBlocks(blockFixtures[block]);
      const channelBlockIds = allBlocks.map((block) => block.id);
      const config: TemplateConfig = { email: { emailTemplateConfig: {} } };
      const template = getProviderTemplate({
        allBlocks,
        channelBlockIds,
        config,
        isEmail: true,
      });
      const emailTemplate = template.emailRenderer();
      const data = {
        data: mockData,
        profile: mockProfile,
        urls: { opened: "https://abc.ct0.app/o/open-tracking" },
      };
      const links: { [context: string]: ILinkData } = {};
      const trackingHandler = getTrackingHandler(links);
      const variableHandler = createVariableHandler({ value: data }).getScoped(
        "data"
      );
      const linkHandler = createLinkHandler(links, true, true, trackingHandler);
      emailTemplate.render(variableHandler, linkHandler);
      expect(links).toMatchSnapshot();
    });
  });
});

describe("email templates", () => {
  const testTemplates = [
    "emailLineTemplateWithHeaderLogo",
    "emailLineTemplateWithFooterLinks",
  ];

  const only = false; // "emailLineTemplateWithFooterTextSocialLinks";

  const allBlocks = hydrateBlocks(blockFixtures.textBlock);
  const channelBlockIds = allBlocks.map((block) => block.id);

  beforeEach(() => (usingTemplate = true));

  testTemplates.forEach((templateName) => {
    const fn = only && only === templateName ? it.only : it;

    fn(`should render ${templateName}`, () => {
      const fixture = templateFixtures[templateName];
      const config = { email: fixture.email };
      const template = getProviderTemplate({
        allBlocks,
        channelBlockIds,
        config,
        isEmail: true,
      });
      const emailTemplate = template.emailRenderer();
      const data = {
        data: mockData,
        profile: mockProfile,
        urls: { opened: "https://abc.ct0.app/o/open-tracking" },
      };
      const links: { [context: string]: ILinkData } = {};
      const trackingHandler = getTrackingHandler(links);
      const variableHandler = createVariableHandler({ value: data }).getScoped(
        "data"
      );
      const linkHandler = createLinkHandler(links, true, true, trackingHandler);
      emailTemplate.render(variableHandler, linkHandler);
      expect(links).toMatchSnapshot();
    });
  });
});
