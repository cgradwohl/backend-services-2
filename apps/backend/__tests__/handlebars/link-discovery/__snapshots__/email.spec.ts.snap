// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email blocks should render actionBlock 1`] = `
Object {
  "$.html.d918bf09-4789-4c57-b930-3c0a5919dcc1.action": Object {
    "context": "$.html.d918bf09-4789-4c57-b930-3c0a5919dcc1.action",
    "options": Object {
      "href": "https://www.courier.com",
      "text": "Click Here",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bm9pdgnhljfjy2q5mtk1ytbjmy0wmzliltc1yzqtotg3nc05mgziode5zc5sbxroliq=",
    "trackingId": "bm9pdgnhljfjy2q5mtk1ytbjmy0wmzliltc1yzqtotg3nc05mgziode5zc5sbxroliq=",
  },
}
`;

exports[`email blocks should render actionBlockLink 1`] = `
Object {
  "$.html.872b59d3-b1db-4ea4-8d3a-bdff41d529be.action": Object {
    "context": "$.html.872b59d3-b1db-4ea4-8d3a-bdff41d529be.action",
    "options": Object {
      "href": "https://www.example.com",
      "text": "link text",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bm9pdgnhlmvioti1zde0zmzkyi1hm2q4ltrhztqtymqxyi0zzdk1yji3oc5sbxroliq=",
    "trackingId": "bm9pdgnhlmvioti1zde0zmzkyi1hm2q4ltrhztqtymqxyi0zzdk1yji3oc5sbxroliq=",
  },
}
`;

exports[`email blocks should render actionBlockWebhook 1`] = `
Object {
  "$.html.9bb67b21-2c98-48b3-ae16-d649f51b45ac.action": Object {
    "context": "$.html.9bb67b21-2c98-48b3-ae16-d649f51b45ac.action",
    "options": Object {
      "href": "https://www.courier.com",
      "text": "Click Here",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bm9pdgnhlmnhntrimtvmotq2zc02mwvhltniodqtodljmi0xmmi3nmjios5sbxroliq=",
    "trackingId": "bm9pdgnhlmnhntrimtvmotq2zc02mwvhltniodqtodljmi0xmmi3nmjios5sbxroliq=",
  },
}
`;

exports[`email blocks should render imageBlock 1`] = `
Object {
  "$.html.a0c2d1d2-a9b6-4997-a6da-22762bd3b7b7.image": Object {
    "context": "$.html.a0c2d1d2-a9b6-4997-a6da-22762bd3b7b7.image",
    "options": Object {
      "href": "https://example.com/mountains",
      "text": "mountains",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/zwdhbwkun2i3yjnkyji2nziylwfknmetnzk5nc02yjlhltjkmwqyyzbhlmxtdgguja==",
    "trackingId": "zwdhbwkun2i3yjnkyji2nziylwfknmetnzk5nc02yjlhltjkmwqyyzbhlmxtdgguja==",
  },
}
`;

exports[`email blocks should render imageBlockWithVars 1`] = `
Object {
  "$.html.a0c2d1d2-a9b6-4997-a6da-22762bd3b7b7.image": Object {
    "context": "$.html.a0c2d1d2-a9b6-4997-a6da-22762bd3b7b7.image",
    "options": Object {
      "href": "https://example.com/{imageLink}",
      "text": "Description: {imageDescription}",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/zwdhbwkun2i3yjnkyji2nziylwfknmetnzk5nc02yjlhltjkmwqyyzbhlmxtdgguja==",
    "trackingId": "zwdhbwkun2i3yjnkyji2nziylwfknmetnzk5nc02yjlhltjkmwqyyzbhlmxtdgguja==",
  },
}
`;

exports[`email blocks should render listBlockImagesWithLinks 1`] = `
Object {
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.0.child-item.0.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.0.child-item.0.rich-text.0",
    "options": Object {
      "href": "https://example.com/lightsaber",
      "text": "lightsaber",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumc5tzxrplwrsawhjljaubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
    "trackingId": "mc50egv0lwhjaxiumc5tzxrplwrsawhjljaubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.0.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.0.rich-text.0",
    "options": Object {
      "href": "https://example.com/luke",
      "text": "Luke Skywalker",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumc5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
    "trackingId": "mc50egv0lwhjaxiumc5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.0.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.0.rich-text.0",
    "options": Object {
      "href": "https://example.com/blaster",
      "text": "blaster",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumc5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
    "trackingId": "mc50egv0lwhjaxiumc5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.1.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.1.rich-text.0",
    "options": Object {
      "href": "https://example.com/z-6-jetpack",
      "text": "Z-6 Jetpack",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiums5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
    "trackingId": "mc50egv0lwhjaxiums5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.2.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.child-item.2.rich-text.0",
    "options": Object {
      "href": "https://example.com/flame-thrower",
      "text": "flame thrower",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumi5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
    "trackingId": "mc50egv0lwhjaxiumi5tzxrplwrsawhjljeubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.1.rich-text.0",
    "options": Object {
      "href": "https://example.com/boba-fett",
      "text": "Boba Fett",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiums5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
    "trackingId": "mc50egv0lwhjaxiums5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.2.child-item.0.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.2.child-item.0.rich-text.0",
    "options": Object {
      "href": "https://example.com/blaster-pistol",
      "text": "blaster pistol",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumc5tzxrplwrsawhjljiubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
    "trackingId": "mc50egv0lwhjaxiumc5tzxrplwrsawhjljiubwv0as45m2u1yzbhyji5zjmtyjliyi0zmzy0ltqxywqtndzjztjhzjuubg10ac4k",
  },
  "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.2.rich-text.0": Object {
    "context": "$.html.5fa2ec64-da14-4633-bb9b-3f92ba0c5e39.item.2.rich-text.0",
    "options": Object {
      "href": "https://example.com/han",
      "text": "Han Solo",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumi5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
    "trackingId": "mc50egv0lwhjaxiumi5tzxrpljkzztvjmgfimjlmmy1iowjiltmznjqtndfhzc00nmnlmmfmns5sbxroliq=",
  },
}
`;

exports[`email blocks should render quoteBlockWithLink 1`] = `
Object {
  "$.html.ec58343f-2c14-4472-b132-94fb451fc7c0.rich-text.0": Object {
    "context": "$.html.ec58343f-2c14-4472-b132-94fb451fc7c0.rich-text.0",
    "options": Object {
      "href": "https://example.com",
      "text": "link",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiumgm3y2yxntrizjq5ltizmwitmjc0nc00mwmylwyzndm4nwnllmxtdgguja==",
    "trackingId": "mc50egv0lwhjaxiumgm3y2yxntrizjq5ltizmwitmjc0nc00mwmylwyzndm4nwnllmxtdgguja==",
  },
}
`;

exports[`email blocks should render templateBlockWithLinks 1`] = `
Object {
  "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.0": Object {
    "context": "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.0",
    "options": Object {
      "href": "https://example.com/0",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
    "trackingId": "mc5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
  },
  "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.1": Object {
    "context": "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.1",
    "options": Object {
      "href": "https://example.com/1",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/ms5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
    "trackingId": "ms5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
  },
  "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.2": Object {
    "context": "$.html.eb16644e-309d-4b6b-90b6-eefc54a466c4.template.2",
    "options": Object {
      "href": "https://example.com/2",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mi5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
    "trackingId": "mi5ldgfscg1ldc40yzy2nge0nwnmzwutnmiwos1inmi0lwq5mdmtztq0njyxymuubg10ac4k",
  },
}
`;

exports[`email blocks should render textBlockWithLink 1`] = `
Object {
  "$.html.a75a20bf-bda9-4c77-9636-90c5079af37b.rich-text.0": Object {
    "context": "$.html.a75a20bf-bda9-4c77-9636-90c5079af37b.rich-text.0",
    "options": Object {
      "href": "https://example.com",
      "text": "Luke",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiuyjczzme5nza1yza5ltyznjktnzdjnc05ywrilwzimdjhntdhlmxtdgguja==",
    "trackingId": "mc50egv0lwhjaxiuyjczzme5nza1yza5ltyznjktnzdjnc05ywrilwzimdjhntdhlmxtdgguja==",
  },
}
`;

exports[`email blocks should render textBlockWithLinks 1`] = `
Object {
  "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.0": Object {
    "context": "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.0",
    "options": Object {
      "href": "https://example.com/1",
      "text": "Link 1",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
    "trackingId": "mc50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
  },
  "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.1": Object {
    "context": "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.1",
    "options": Object {
      "href": "https://example.com/2",
      "text": "Link 2",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/ms50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
    "trackingId": "ms50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
  },
  "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.2": Object {
    "context": "$.html.4478b6ae-8e6e-4da3-b4af-b091bb748e74.rich-text.2",
    "options": Object {
      "href": "https://example.com/3",
      "text": "Link 3",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mi50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
    "trackingId": "mi50egv0lwhjaxiunddlodq3ymixotbilwzhngitm2fknc1lnmu4lwvhnmi4nzq0lmxtdgguja==",
  },
}
`;

exports[`email templates should render emailLineTemplateWithFooterLinks 1`] = `
Object {
  "$.html.email-footer.facebook": Object {
    "context": "$.html.email-footer.facebook",
    "options": Object {
      "href": "https://fb.me/trycourier",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/a29vymvjywyucmv0b29mlwxpyw1llmxtdgguja==",
    "trackingId": "a29vymvjywyucmv0b29mlwxpyw1llmxtdgguja==",
  },
  "$.html.email-footer.instagram": Object {
    "context": "$.html.email-footer.instagram",
    "options": Object {
      "href": "https://www.instagram.com/trycourier/",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bwfyz2f0c25plnjldg9vzi1sawftzs5sbxroliq=",
    "trackingId": "bwfyz2f0c25plnjldg9vzi1sawftzs5sbxroliq=",
  },
  "$.html.email-footer.linkedin": Object {
    "context": "$.html.email-footer.linkedin",
    "options": Object {
      "href": "https://www.linkedin.com/company/trycourier/",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bmlkzwtuawwucmv0b29mlwxpyw1llmxtdgguja==",
    "trackingId": "bmlkzwtuawwucmv0b29mlwxpyw1llmxtdgguja==",
  },
  "$.html.email-footer.medium": Object {
    "context": "$.html.email-footer.medium",
    "options": Object {
      "href": "https://medium.com",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/bxvpzgvtlnjldg9vzi1sawftzs5sbxroliq=",
    "trackingId": "bxvpzgvtlnjldg9vzi1sawftzs5sbxroliq=",
  },
  "$.html.email-footer.rich-text.0": Object {
    "context": "$.html.email-footer.rich-text.0",
    "options": Object {
      "href": "https://example.com/",
      "text": "link",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/mc50egv0lwhjaxiucmv0b29mlwxpyw1llmxtdgguja==",
    "trackingId": "mc50egv0lwhjaxiucmv0b29mlwxpyw1llmxtdgguja==",
  },
  "$.html.email-footer.twitter": Object {
    "context": "$.html.email-footer.twitter",
    "options": Object {
      "href": "https://twitter.com/trycourier",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/cmv0dgl3dc5yzxrvb2ytbglhbwuubg10ac4k",
    "trackingId": "cmv0dgl3dc5yzxrvb2ytbglhbwuubg10ac4k",
  },
}
`;

exports[`email templates should render emailLineTemplateWithHeaderLogo 1`] = `
Object {
  "$.html.email-header.logo": Object {
    "context": "$.html.email-header.logo",
    "options": Object {
      "href": "https://courier.com",
    },
    "renderCount": 1,
    "trackingHref": "https://abc-tenant-id.ct0.app/r/b2dvbc5yzwrhzwgtbglhbwuubg10ac4k",
    "trackingId": "b2dvbc5yzwrhzwgtbglhbwuubg10ac4k",
  },
}
`;
